packages/mds-db/policies.ts

```diff
@@ -56,6 +56,22 @@ export async function readPolicies(params?: {
   return res.rows.map(row => row.policy_json)
 }
 
+export async function readActivePolicies(timestamp: Timestamp = now()): Promise<Policy[]> {
```

Comment:
I don't see why we can't just reuse the `readPolicies` method here, but if there's a need to create a new method then the naming seems a little off -- when I think of active policies I think of `start_date <= now && end_date >= now && published` roughly (with the requirement for some null checking as well). It seems that the start/end restrictions are being met, but it's not considering if the policies are published or not

Reply:
It is checking for a non-null `publish_date` on line 79.

Reply:
but publish_date should take into account the timestamp

Reply:
I think that was not the case when I made this comment ;) I think the fundamental issue I have with this is that we're using a different method for determining "active policies" between `Policy` and `Compliance`. They really ought to be using the same methodology, which is tricky because `Policy` enables the user to get non-active policies depending on their permissions.

As much as I'd like to do all of the filtering in the DB query, I think what may make sense is to just use the same logic that we have for `Policy`
```
const policies = await db.readPolicies({ get_published, get_unpublished })
      const prev_policies: UUID[] = policies.reduce((prev_policies_acc: UUID[], policy: Policy) => {
        if (policy.prev_policies) {
          prev_policies_acc.push(...policy.prev_policies)
        }
        return prev_policies_acc
      }, [])
      const active = policies.filter(p => {
        // overlapping segment logic
        const p_start_date = p.start_date
        const p_end_date = p.end_date || Number.MAX_SAFE_INTEGER
        return end_date >= p_start_date && p_end_date >= start_date && !prev_policies.includes(p.policy_id)
      })
```

Reply:
(Indentation got substantially borked, but you get the idea)